packages/examples/src/scenes/components-switch.tsx

```diff
@@ -0,0 +1,85 @@
+import {Circle, Node, NodeProps, Rect} from '@motion-canvas/2d/lib/components';
+import {easeInOutCubic, tween} from '@motion-canvas/core/lib/tweening';
+import {
+  Color,
+  ColorSignal,
+  PossibleColor,
+} from '@motion-canvas/core/lib/types/Color';
+import {initial, signal} from '@motion-canvas/2d/lib/decorators';
+import {
+  createSignal,
+  SignalValue,
+  SimpleSignal,
+} from '@motion-canvas/core/lib/signals';
+import {createRef} from '@motion-canvas/core/lib/utils';
+import {all} from '@motion-canvas/core/lib/flow';
+
+export interface SwitchProps extends NodeProps {
+  initialState?: SignalValue<boolean>;
+  accent?: SignalValue<PossibleColor>;
+}
+
+export class Switch extends Node {
+  @initial(false)
+  @signal()
+  public declare readonly initialState: SimpleSignal<boolean, this>;
+
+  @initial('#68ABDF')
+  @signal()
+  public declare readonly accent: ColorSignal<this>;
+
+  private isOn: boolean;
+  private readonly indicatorPosition = createSignal(0);
+  private readonly offColor = new Color('#242424');
+  private readonly indicator = createRef<Circle>();
+  private readonly container = createRef<Rect>();
+
+  public constructor(props?: SwitchProps) {
+    super({
```

Comment:
Super minor detail: Is there a reason why you create a new object and spread the `props` instead of just passing the props directly?

```ts
super(props);
```

Reply:
This and the the empty `<>` when adding only a single root element also apply to a few more places in this file.

Reply:
> Super minor detail: Is there a reason why you create a new object and spread the `props` instead of just passing the props directly?
> 
> ```ts
> super(props);
> ```

because if you were to make a layout component and wanted to ensure it was a layout, you would use 
```tsx
super({
   layout
   ...props
})
```

Reply:
I'm not sure I'm following. Where is the `layout` variable coming from in this example?

Reply:
because if you were to make a layout component and wanted to ensure it was a layout, you would use :
```tsx
super({
   layout
   ...props
})
```
just makes it a bit more obvious not only props can go here

Reply:
If you were making a layout component, you'd also be extending from `Layout` and not `Node`, however. So the props in this case would also extends `LayoutProps` and not `NodeProps`. At which point `super(props)` would still work as expected.

Reply:
Yeah no I get that but was confused because in that concrete example it is not needed. But if you want to add it just to provide a generic way to make it easier understandable for someone that isn't so much into typescript / coding I am all for it. But providing an example / paragraph describing this could be useful then as well.

Not sure however at what persona the documentation is targeted. Maybe @aarthificial could shed some light on this.